src/vs/workbench/contrib/notebook/browser/notebookExecutionStateServiceImpl.ts

```diff
@@ -139,11 +140,45 @@ export class NotebookExecutionStateService extends Disposable implements INotebo
 			exe.onDidComplete(lastRunSuccess => this._onCellExecutionDidComplete(notebookUri, cellHandle, exe, lastRunSuccess)));
 		this._cellListeners.set(CellUri.generate(notebookUri, cellHandle), disposable);
 
+		notebook.onWillAddRemoveCells((e: NotebookTextModelWillAddRemoveEvent) => {
+			const lastFailedCell = this.getLastFailedCellForNotebook(notebook.uri);
+			e.rawEvent.changes.forEach(([start, deleteCount]) => {
```

Comment:
Rather than `changes.forEach` you could use `changes.some` which would let this loop terminate early if _some_ change removed the cell you are looking for.